internal/net/grpc/client.go

```diff
@@ -210,24 +218,40 @@ func (g *gRPCClient) StartConnectionMonitor(ctx context.Context) (<-chan error,
 								ech <- err
 							}
 						}
-						if err == nil {
+						if err == nil && addr != "" && p != nil {
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg?style=flat-square)
`addr != ""` is already checked at L209 (`len(addr) != 0`) and `addr` is not modified between L209 and L221.